### PR TITLE
Code39 reading support

### PR DESCRIPTION
**Pull request by jojastahl**
I ported Code39Reader from Java including tests.

### CODE REVIEW

File: webpack.config.test.js [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
+        entry: './src/index.ts',
+        resolve: {
+            extensions: ['.ts'],
+        },
+        module: {
+            rules: [{
+                test: /\.ts$/,
+                exclude: /node_modules/,
+                include: path.resolve('src'), // instrument only testing sources with Istanbul, after ts-loader runs
+                use: ifDebug('ts-loader', ['istanbul-instrumenter-loader', 'ts-loader'])
```
Comment (odahcam) [inline webpack.config.test.js]: I recommend to use Webpack 4 modes instead.

File: src/core/oned/MultiFormatOneDReader.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ import BitArray from './../common/BitArray';
 import ITFReader from './ITFReader';
 import MultiFormatUPCEANReader from './MultiFormatUPCEANReader';
 import NotFoundException from '../NotFoundException';
+import Code39Reader from './Code39Reader';
```
Comment (odahcam) [inline src/core/oned/MultiFormatOneDReader.ts]: Let's order this alphabetically.

File: webpack.config.test.js (hunk 1 above)
Comment (jojastahl) [reply webpack.config.test.js]: As far as I can see, mode only supports `development` and `production`: https://webpack.js.org/concepts/mode/

But for this case I wanted to have different ways for `development`. The reason is the `istanbul-instrumenter-loader`. If it is active, breakpoints don't work any more due to issues with source maps.

File: src/core/oned/MultiFormatOneDReader.ts (hunk 2 above)
Comment (jojastahl) [reply src/core/oned/MultiFormatOneDReader.ts]: These imports aren't ordered at all. And there are two groups:

https://github.com/jojastahl/library/blob/2b7f0470699aa5593880775ee8f62ab1fed6525c/src/core/oned/MultiFormatOneDReader.ts#L19-L29

Shall I put the two groups together and order all alphabetically? Or in which group would you like to see Code39Reader?

Comment (odahcam) [reply src/core/oned/MultiFormatOneDReader.ts]: I took a look at the ZXing original code and I can confirm what you're saying. Let's make our ordering for now by putting the imports in alphabetic order. 🙂

File: webpack.config.test.js (hunk 1 above)
Comment (odahcam) [reply webpack.config.test.js]: Ok, let's continue with your implementation then. I can't provide any better options at the moment.

**Pull request merged**